### PR TITLE
Changed CI and use of buld arg to pass. commit SHA

### LINKED ISSUE #25: Synchronize Pharo test runner image with Pharo-Smalltalk repository
**Problem statement**
There is a need to rebuild Pharo image for test runner, in order to use latest version of Pharo-smalltalk repository, where exercises are defined. Otherwise test runner image doesn't contain latest changes and tests cannot be ran on latest exercises.
Therefore new fresh docker image with latest changes has to be rebuilt. Unfortunately, docker build action process caches the unchanged line of Dockerfile and thus using older artifacts of build process.
Evidence here: 

**Proposed soluton**
Enforce line reponsible for creating new Pharo image for test runner, and pass there latest commit SHA of Pharo-smalltalk repository as build argument. 
Build process will rebuild the image, since line has been changed (by using passed SHA)

Build argument SHA_OR_BRANCH is defined as env.variable speficied in CI.yml specification of GH action. 

**Related discussion:** [Forum: test-runner-update-in-docker](http://forum.exercism.org/t/test-runner-update-in-docker-results-interpretation-by-ui/7344/10)

### PR DESCRIPTION
**Pull request by Bajger**
- assign env variable from file in GH action
- pass build-args to Docker build step in GH action
- use build argument in dockerfile
- fixes #25

### CODE REVIEW

**Comment by Bajger** [comment]
@ErikSchierboom For some reason CI GH action didn't run. Do I have some error in CI.yml?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,6 +21,14 @@ jobs:
         with:
           install: true
 
+      - name: Read content of Pharo-sync-commit file and set its 3rd line to environment variable (ignoring comments)
+      run: |
+        # Read the content of the second line of the file and store it in SHA_OR_BRANCH
+        SHA_OR_BRANCH=$(sed -n '3p' Pharo-sync-commit)
+        echo "SHA_OR_BRANCH=$SHA_OR_BRANCH" >> $GITHUB_ENV
+      working-directory: ${{ github.workspace }}
+
```
Comment (ErikSchierboom) [inline .github/workflows/ci.yml]: I don't think you need this (it's the default)

```suggestion
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -21,6 +21,14 @@ jobs:
         with:
           install: true
 
+      - name: Read content of Pharo-sync-commit file and set its 3rd line to environment variable (ignoring comments)
+      run: |
+        # Read the content of the second line of the file and store it in SHA_OR_BRANCH
+        SHA_OR_BRANCH=$(sed -n '3p' Pharo-sync-commit)
+        echo "SHA_OR_BRANCH=$SHA_OR_BRANCH" >> $GITHUB_ENV
```
Comment (ErikSchierboom) [inline .github/workflows/ci.yml]:
```suggestion
        echo "SHA_OR_BRANCH=${SHA_OR_BRANCH}" >> $GITHUB_ENV
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -30,6 +38,8 @@ jobs:
           tags: exercism/pharo-smalltalk-test-runner
           cache-from: type=gha
           cache-to: type=gha,mode=max
+          build-args: |
+            SHA_OR_BRANCH=${{ env.SHA_OR_BRANCH }}
```
Comment (ErikSchierboom) [inline .github/workflows/ci.yml]:
```suggestion
          build-args: SHA_OR_BRANCH=${{ env.SHA_OR_BRANCH }}
```

**Review by ErikSchierboom — COMMENTED** [review]
Nice work! Couple of minor nits

**Comment by Bajger** [comment]
1. It seems that build arg is correnctly passed in Build docker image step: https://github.com/exercism/pharo-smalltalk-test-runner/actions/runs/6259811474/job/16996565397#step:5:123
2. And later I see that docker run correctly evaluate that: https://github.com/exercism/pharo-smalltalk-test-runner/actions/runs/6259811474/job/16996565397#step:5:218

**Comment by Bajger** [comment]
@ErikSchierboom: If you agree and approve golden tests change in prior PR, CI should be green 🎉

**Comment by ErikSchierboom** [comment]
@Bajger Great! The other PR has failing CI: https://github.com/exercism/pharo-smalltalk-test-runner/actions/runs/6246090121/job/16956001500?pr=23

**Comment by Bajger** [comment]
@ErikSchierboom Yes the other PR is failing, since it expects changes from fresh build of Pharo image coming from this PR :)

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**